zigpy/util.py

```diff
@@ -286,6 +286,8 @@ async def catching_coro(
             return await target
         except exceptions:
             pass
+        except asyncio.CancelledError:
+            raise
```

Comment:
should we raise here or just handle the same way as the broad Exception? For catching coros that was the intention, not to raise any exceptions so you don't get "exception was never retrieved"

Reply:
It doesn't appear that an error is logged when either the `catching_coro` task or an inner task is cancelled:

```Python
import asyncio


async def catching_coro(target, exceptions):
    try:
        return await target
    except exceptions:
        pass
    except asyncio.CancelledError:  # For Python 3.7
        raise
    except Exception as e:
        print(f"Caught exception: {e}")


async def busy():
    while True:
        print('Busy')
        await asyncio.sleep(0.5)

async def main():
    busy_task = asyncio.create_task(busy())
    catching_task = asyncio.create_task(catching_coro(busy_task, (RuntimeError,)))

    await asyncio.sleep(1)

    print('Cancelling')
    busy_task.cancel()
    catching_task.cancel()

    await asyncio.sleep(1)
    print('Done')


if __name__ == '__main__':
    asyncio.run(main())
```

Reply:
Ignore that, it does get logged if you `await catching_task`. It doesn't if you catch the `CancelledError`.

Yeah, maybe here it does make sense to catch both here?